public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -52,8 +99,33 @@ export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
           if (!QueryEditor) {
             return <Alert title="Data source does not export a query editor."></Alert>;
           }
-
-          return <QueryEditor onRunQuery={() => {}} onChange={onChange} datasource={datasource} query={value} />;
+          return (
+            <>
+              <QueryEditor
+                onRunQuery={() => handleValidation(value)}
+                onChange={(value) => {
+                  setIsValidQuery(undefined);
+                  onChange(value);
+                }}
+                datasource={datasource}
+                query={value}
+              />
+              <HorizontalGroup justify="flex-end">
+                {isValidQuery ? (
+                  <div className={styles.valid}>
+                    <Icon name="check" /> This query is valid.
+                  </div>
+                ) : isValidQuery === false ? (
+                  <FieldValidationMessage>This query is not valid.</FieldValidationMessage>
+                ) : (
+                  <div />
+                )}
```

Comment:
do we really need to render that empty div?
```suggestion
                ) : null}
```